internal/mode/static/state/graph/route_common_test.go

```diff
@@ -1584,3 +1734,61 @@ func TestRouteKeyForKind(t *testing.T) {
 
 	g.Expect(rk).To(Panic())
 }
+
+func TestAllowedRouteType(t *testing.T) {
+	test := []struct {
+		listener  *Listener
+		name      string
+		routeType RouteType
+		expResult bool
+	}{
+		{
+			name:      "grpcRoute attaches to listener with allowedRoutes set to grpcRoute",
```

Comment:
nit: let's change these test messages a bit to better reflect the behavior of the function
```suggestion
			name:      "grpcRoute is allowed when listener supports grpcRoute kind ",
```

same for the other tests 🙂